openedx/openedx-platform
# [BD-13] Deprecates remaining user-related properties on ModuleSystem

## Description

Builds on the shim added by https://github.com/edx/edx-platform/pull/29190 to deprecate these remaining user-related `ModuleSystem` attributes in favour of the user service:

* user_location
* get_real_user
* get_user_role

These properties are only used by a few XBlocks: LTI, LTI2.0, Video (CDN/branding), edxnotes

This change is only a refactoring, and should not affect Learners, Course Authors, or anyone else using edx-platform.

## Supporting information

* [Content Core Platform Simplication project](https://openedx.atlassian.net/wiki/spaces/AC/pages/3091629503/Content+Core+Platform+Simplification)
* [BLENDED-113](https://openedx.atlassian.net/browse/BLENDED-113) (edX internal ticket)
* [FAL-2399](https://tasks.opencraft.com/browse/FAL-2399) (OpenCraft internal ticket)

## Testing instructions

* LMS: https://pr29375.sandbox.opencraft.hosting/
* Studio: https://studio.pr29375.sandbox.opencraft.hosting/

Basic regression testing can be run on the sandbox.

To test the LTI 2.0 (consumer) component: 
1. View the [LTI component in Studio](https://studio.pr29375.sandbox.opencraft.hosting/container/block-v1:OpenCraft+BD-13+2021_1+type@vertical+block@0a1f09b607794a42adefde02b49184f4).
   Note that the LTI URL points to a problem block on a different sandbox server's Demo course.
 1.  Ensure you can edit the component, preview it, view it in the LMS, and submit answers to the problems shown.

To test edx-notes:

1. Login to LMS as a student user.
1. View the BD-13 test course in the LMS.
2. Navigate to an [HTML block](https://discovery.pr29375.sandbox.opencraft.hosting/learning/course/course-v1:OpenCraft+BD-13+2021_1/block-v1:OpenCraft+BD-13+2021_1+type@sequential+block@c2883d06799c4aaf845d6abd4f934328/block-v1:OpenCraft+BD-13+2021_1+type@vertical+block@a516543787854cb2903ffd1c2a642f8e), select some text, and ensure you can add/edit notes.
3. Toggle the hide/show notes button in the lower right corner to hide/show notes.
2. Click on the [Notes](https://pr29375.sandbox.opencraft.hosting/courses/course-v1:OpenCraft+BD-13+2021_1/edxnotes/) tab and ensure your notes are visible there.

To test the video CDN/branding:

You need to view the site from one of the CDN-configured countries, currently Australia (`AU`), Pakistan (`PK`), or Poland (`PL`).

Note: I don't really have a valid CDN to use here, but enabling a CDN allows a [configured branding logo to be shown under the video](https://github.com/edx/edx-platform/blob/8b3d7e040c2d3b138e443466c83ed3dc7adcb94a/common/lib/xmodule/xmodule/video_module/video_module.py#L336-L337), so this at least is visible and can be tested.

1. View the [test Video in the LMS](https://pr29375.sandbox.opencraft.hosting/courses/course-v1:OpenCraft+BD-13+2021_1/jump_to/block-v1:OpenCraft+BD-13+2021_1+type@vertical+block@3f7e6c9b074c4bb7b4a2f6c7aa6b1d2c).
   Note that the configured branding logo is shown under the video when you view this content from one of the configured countries.
 
## Deadline

None

## Author Notes & Concerns

Setup for user location/CDN testing:

1. Update `EDXAPP_VIDEO_CDN_URLS` or `/edx/etc/lms.yml` to provide a CDN URL for a given country code (doesn't matter what the URL is), eg.
   ```yaml
   VIDEO_CDN_URL:
     AU: http://example.com/edx/video?s3_url=
   ```
1. As a superuser, navigate to Django Admin > Branding > [Branding Info Config](https://pr29375.sandbox.opencraft.hosting/admin/branding/brandinginfoconfig/), and add a branding url, logo, and logo_tag for your country, e.g.
   ```json
   {
      "AU": {
        "url": "https://www.sa.gov.au",
        "logo_src": "https://www.sa.gov.au/__data/assets/image/0012/251013/sa-logo.png",
        "logo_tag": "Logo for South Australia"
      }
    }
    ```
